### PR TITLE
Add support for range verfication

### PR DESCRIPTION
**Pull request by afcidk**
Add `trange` parameter to `hash_chain_verify` to specify the maximum range.

The default test range is specified in `DEFAULT_TRANGE`.

### CODE REVIEW

File: hashchain.c [hunk 1]
```diff
@@ -64,24 +65,37 @@ struct hash_chain hash_chain_create(void *base,
  * @param h Pointer to a hash.
  * @param tip Pointer to the "tip" hash.
  * @param hash Hash algorithm to use.
- * @param True if hash(h) == tip
+ * @param trange Maximum test size.
```
Comment (jserv) [inline hashchain.c]: Why would you call it `trange`?

Comment (afcidk) [reply hashchain.c]: I think the semantics of this function is "Could the function be verified in the given range?".

Comment (jserv) [reply hashchain.c]: Alternatively, just use `range`.

**Comment by jserv** [comment]
Please update `README.md` accordingly.

File: hashchain.c [hunk 2]
```diff
@@ -161,11 +175,12 @@ int cmd_verify(int argc, char **argv)
 {
     if (argc < 4) {
         fprintf(stderr, "error: too few args\n");
-        fprintf(stderr, "usage: %s ALGO QUERY ANCHOR\n", argv[0]);
+        fprintf(stderr, "usage: %s ALGO QUERY ANCHOR [MAX_TRANGE]\n", argv[0]);
```
Comment (jserv) [inline hashchain.c]: Shall it be `MAX_RANGE`?

**Pull request merged**